api-update/antivirus/src/test/scala/uk/gov/nationalarchives/api/update/antivirus/AntivirusAntivirusUpdateTest.scala

```diff
@@ -0,0 +1,101 @@
+package uk.gov.nationalarchives.api.update.antivirus
+
+import java.net.URI
+
+import com.github.tomakehurst.wiremock.client.WireMock.{equalTo, equalToJson, postRequestedFor, urlEqualTo}
+import org.scalatest.matchers.should.Matchers._
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sqs.SqsClient
+import software.amazon.awssdk.services.sqs.model.{ReceiveMessageRequest, SendMessageRequest}
+import uk.gov.nationalarchives.api.update.common.utils.AWSInputs._
+import uk.gov.nationalarchives.api.update.common.utils.ExternalServicesTest
+
+import scala.io.Source.fromResource
+import scala.util.Try
+
+class AntivirusAntivirusUpdateTest extends ExternalServicesTest {
+
+  def verifyWiremockResponse(fileName: String) = {
+    wiremockGraphqlServer.verify(postRequestedFor(urlEqualTo(graphQlPath))
+      .withRequestBody(equalToJson(fromResource(s"json/$fileName.json").mkString))
+      .withHeader("Authorization", equalTo("Bearer token")))
+  }
+
+  "The update method" should "call the graphql api with a single record with a single antivirus update" in {
+    authOkJson("access_token")
+    graphqlOkJson("graphql_valid_av_response")
+    val main = new AntivirusUpdate()
+    main.update(sqsEvent("function_valid_av_input"), context)
+    verifyWiremockResponse("graphql_valid_av_expected")
+  }
+
+  "The update method" should "call the graphql api with a single record with multiple antivirus updates" in {
+    authOkJson("access_token")
+    graphqlOkJson("graphql_valid_av_response")
+    new AntivirusUpdate().update(sqsEvent("function_valid_av_multiple_input"), context)
```

Comment:
Does this mean there are multiple updates in the same message? Does this suffer from the same problem as updating the API with multiple results at the same time? i.e. if one fails, the whole message gets put back on the queue?

Reply:
Yeah, you're right. I've changed the antivirus python lambda to send one update per sqs message so we'll only get one update per record in the api-update lambda.